rewrite-java-25/src/main/java/org/openrewrite/java/isolated/ReloadableJava25ParserVisitor.java

```diff
@@ -846,10 +846,29 @@ public J visitDeconstructionPattern(DeconstructionPatternTree node, Space fmt) {
                 type);
     }
 
+    @Override
+    public J visitAnyPattern(AnyPatternTree node, Space space) {
+        JavaType type = typeMapping.type(node);
+        skip("_");
+        J.Identifier name = new J.Identifier(randomId(), space, Markers.EMPTY, emptyList(), "_",
+                type instanceof JavaType.Variable ? ((JavaType.Variable) type).getType() : type,
+                type instanceof JavaType.Variable ? (JavaType.Variable) type : null);
+        if (name.getFieldType() != null && name.getFieldType().getName().isEmpty()) {
+            name = name.withFieldType(name.getFieldType().withName("_"));
```

Comment:
I can't get the tests to run locally yet to attach a debugger, but it reads a bit odd to me to create an instance with a specific type and field type, and then directly following change only the field type on the created instance.

Would it not make more sense to either change the `type` variable on line 851 directly, and thereby remove the conditional and partial update, or even move that into `typeMapping.type(node)` for unnamed variables anywhere?